src/base/algorithm.h

```diff
@@ -28,14 +28,42 @@
 
 #pragma once
 
-namespace Dict
+#include <type_traits>
+
+namespace Algorithm
 {
-    // To be used with QMap, QHash and it's variants
-    template <typename Dictionary, typename BinaryPredicate>
-    void removeIf(Dictionary &&dict, BinaryPredicate p)
+    template <typename ...>
+    using void_t = void;  // replace this with std::void_t in C++17
+
+    template <typename T, typename = void>
+    struct HasMappedType
+        : std::false_type
+    {
+    };
+
+    template <typename T>
+    struct HasMappedType<T, void_t<typename T::mapped_type>>
+        : std::true_type
+    {
+    };
+
+    // To be used with usual iteratable types, such as QVector, QSet, etc.
+    template <typename T, typename BinaryPredicate
```

Comment:
BinaryPredicate isn't valid name in this case. It should be UnaryPredicate since it is function taking only one parameter.

Reply:
Thanks, fixed.